pkg/apis/storage/validation/validation.go

```diff
@@ -419,8 +419,6 @@ func ValidateCSIDriverUpdate(new, old *storage.CSIDriver) field.ErrorList {
 
 	// immutable fields should not be mutated.
 	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(new.Spec.AttachRequired, old.Spec.AttachRequired, field.NewPath("spec", "attachedRequired"))...)
-	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(new.Spec.FSGroupPolicy, old.Spec.FSGroupPolicy, field.NewPath("spec", "fsGroupPolicy"))...)
```

Comment:
This may be tricky to do, we don't know when a new CSIDriver reaches kubelet's infomer and thus when it's safe to create a Pod checking the new behavior. The field is used only when starting a pod + mounting its volumes.

Reply:
Generally in non-deterministic cases, you poll for some time until you reach the desired state.  I believe in this case you would keep creating pods in a `wait` loop until you saw the new behavior or timed out.

Reply:
e2e test cases added, the test flow is:
1. create csidriver
2. create pod
3. verification
4. update csidriver
5. create pod
6. verification